src/index.ts

```diff
@@ -1,19 +1,649 @@
+   * @param plaintext The data in string format to sign.
+   * @returns The encrypted data.
+   */
+  async encrypt(plaintext: string | BinaryDataType): Promise<Uint8Array> {
+    const sub = this.auth0.getCachedUserSub();
+
+    if (!sub) throw new Error("Missing cached user.");
+
+    const encryptedData = await this.api.encrypt(plaintext, sub);
+
+    return stringToUint8Array(encryptedData);
+  }
+
+  /**
+   * Decrypt data with the users JWK. This function assumes (and requires) a user is logged in and a valid encrypt() response.
+   * @param ciphertext The data to decrypt.
+   * @returns The decrypted data.
+   */
+  async decrypt(ciphertext: string | BinaryDataType): Promise<string> {
+    const sub = this.auth0.getCachedUserSub();
+
+    if (!sub) throw new Error("Missing cached user.");
+
+    const decryptedData = await this.api.decrypt(ciphertext, sub);
+
+    return decryptedData;
+  }
+
+  // SIGN:
+
+  // TODO: Add deprecation warning (and update all TSDocs according to what's on ArConnect and add references to their docs).
+
+  /**
+   * Generate a signature. This function assumes (and requires) a user is logged in.
+   * @param data The data to sign.
+   * @returns The {@linkcode Buffer} format of the signature.
+   */
+  async signature(data: string | BinaryDataType): Promise<Uint8Array> {
+    const sub = this.auth0.getCachedUserSub();
+
+    if (!sub) throw new Error("Missing cached user.");
+
+    const signature = await this.api.sign(data, sub);
+
+    const rawSignature = stringToUint8Array(signature);
+
+    return rawSignature;
+  }
+
+  /**
+   * The signDataItem() function allows you to create and sign a data item object, compatible with arbundles. These data items can then be submitted to an ANS-104 compatible bundler.
+   * @param dataItem The data to sign.
+   * @returns The signed data item.
+   */
+  async signDataItem(dataItem: DataItem): Promise<Buffer> {
+    const publicKeyBuffer = this.auth0.getCachedUserPublicKeyBuffer();
+    const sub = this.auth0.getCachedUserSub();
+
+    if (!publicKeyBuffer || !sub) throw new Error("Missing cached user.");
+
+    const { data, ...options } = dataItem;
+
+    const signer: Signer = {
+      publicKey: publicKeyBuffer,
+      signatureType: 1,
+      signatureLength: 512,
+      ownerLength: 512,
+      sign: this.api.getSignerSignFn(sub),
+      // Note we don't provide `verify` as it's not used anyway:
+      // verify: null,
+    };
+
+    const opts: DataItemCreateOptions = {
+      ...options,
+      tags: [...(options.tags || []), ...ANALYTICS_TAGS],
+    };
+
+    const dataItemInstance = createData(data, signer, opts);
+
+    // DataItem.sign() sets the DataItem's `id` property and returns its `rawId`:
+    await dataItemInstance.sign(signer);
+
+    // TODO: ArConnects types the return type as ArrayBuffer, but in the example this goes straight into the DataItem constructor, which only accepts Buffer.
```

Comment:
@7i7o This is the main discrepancy with ArConnect types:

```
Property 'signDataItem' in type 'Othent' is not assignable to the same property in base type 'Omit<ArConnect, "connect">'.
  Type '(dataItem: DataItem) => Promise<Buffer>' is not assignable to type '(dataItem: DataItem) => Promise<ArrayBufferLike>'.
    Type 'Promise<Buffer>' is not assignable to type 'Promise<ArrayBufferLike>'.
      Type 'Buffer' is not assignable to type 'ArrayBufferLike'.
        Property 'resize' is missing in type 'Buffer' but required in type 'ArrayBuffer'.
```

ArConnects specifies the return type of this as `ArrayBuffer`, so the last statement should be:

    return dataItemInstance.getRaw().buffer;

However, in their docs there's an example where they do:

````
// sign the data item
const signed = await window.arweaveWallet.signDataItem({
    data: "This is an example data",
    tags: [{
        name: "Content-Type",
        value: "text/plain"
    }]
});

// load the result into a DataItem instance
const dataItem = new DataItem(signed);
```

The only way for this to work with TS is to return a `Buffer` instead (how the code is right now). You can see that by uncommenting this line:

   new DataItem(dataItemInstance.getRaw().buffer);